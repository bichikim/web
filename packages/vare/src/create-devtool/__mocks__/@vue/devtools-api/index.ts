export const setupDevtoolsPlugin = (options: any, setup: any) => {
  const api = {
    addInspector: (options) => {
      // empty
    },

    on: {
      editInspectorState: (handler: any) => {
        // empty
      },
      getInspectorState: (handler: any) => {
        // empty
      },
      getInspectorTree: (handler: any) => {
        // empty
      },
    },
  }

  setup(api)
}
