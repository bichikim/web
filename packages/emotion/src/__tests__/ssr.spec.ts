/**
 * @jest-environment node
 */

import {createEmotion} from '../index'
import {createSSRApp, h} from 'vue-demi'
import {renderToString} from '@vue/server-renderer'
import createEmotionServer from '@emotion/server/create-instance'

describe('server-side rendering', () => {
  it('should ', async () => {
    const emotion = createEmotion()
    const emotionServer = createEmotionServer(emotion.cache)

    const Text = emotion.styled('div')({
      color: 'red',
    })

    const app = createSSRApp({
      setup() {
        return () => h('div', [h(Text, () => 'foo')])
      },
    })

    const appContent = await renderToString(app)

    const {html, styles} = emotionServer.extractCriticalToChunks(appContent)
    const stylesInHead = emotionServer.constructStyleTagsFromChunks({html, styles})

    expect(stylesInHead).toMatchSnapshot()
    expect(html).toMatchSnapshot()
  })
})
