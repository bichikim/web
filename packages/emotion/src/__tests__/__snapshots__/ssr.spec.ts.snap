// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server-side rendering should  1`] = `"<style data-emotion=\\"css tokvmb\\">.css-tokvmb{color:red;}</style>"`;

exports[`server-side rendering should  2`] = `"<div><div class=\\"css-tokvmb\\">foo</div></div>"`;
